src/components/apps/savedLaunch/SavedLaunchListing.js

```diff
@@ -218,50 +252,68 @@ function ListSavedLaunches(props) {
         }
     );
 
-    const embedCodeClickHandler = () => {
-        const shareUrl = getShareUrl(selected.id);
+    const { mutate: createInstantLaunch } = useMutation(addInstantLaunch, {
+        onSuccess: (createdIL, { onSuccess }) => {
+            queryClient.invalidateQueries(ALL_INSTANT_LAUNCHES_KEY);
+            setSelectedIL(createdIL);
+            setUseInstantLaunch(true);
+        },
+        onError: setCreateILError,
+    });
+
+    const getShareUrl = useCallback(() => {
+        const host = getHost();
+        const url = `${host}${getSavedLaunchPath(
+            systemId,
+            appId,
+            selected?.id
+        )}`;
+        return url;
+    }, [selected, systemId, appId]);
+
+    useEffect(() => {
+        var shareUrl = "";
         const host = getHost();
         const imgSrc = `${host}/${constants.SAVED_LAUNCH_EMBED_ICON}`;
 
+        if (useInstantLaunch && selectedIL?.id) {
+            shareUrl = `${host}${getInstantLaunchPath(selectedIL?.id)}`;
+        } else {
+            shareUrl = getShareUrl();
+        }
+
         const embed = `<a href="${shareUrl}" target="_blank" rel="noopener noreferrer"><img src="${imgSrc}"></a>`;
 
-        setAnchorEl(null);
         setEmbedCode(embed);
+    }, [selected, selectedIL, useInstantLaunch, getShareUrl]);
+
+    const embedCodeClickHandler = () => {
+        setAnchorEl(null);
         setEmbedDialogOpen(true);
     };
 
     const shareClickHandler = () => {
         setAnchorEl(null);
-        setSavedLaunchUrl(getShareUrl(selected.id));
```

Comment:
`getShareUrl` doesn't actually take an argument, it looks at `selected` itself. So I removed that.